paramnb/__init__.py

```diff
@@ -310,3 +322,58 @@ def format_name(pname):
             widgets.append(display_button)
 
         return widgets
+
+
+class EnvironmentInit(param.ParameterizedFunction):
+    """
+    Callable that can be passed to Widgets.initializer to set Parameter
+    values based on environment variables encoding JSON.
+
+    Here is an easy example of setting such an environment variable on
+    the commandline:
+
+    PARAMNB_INIT='{"p1":5}' jupyter notebook
+
+    This addresses any EnvironmentInit instances that are inspecting the
+    default environment variable called PARAMNB_INIT, instructing it to set
+    the 'p1' parameter to 5.
+    """
+
+    varname = param.String(default='PARAMNB_INIT', doc="""
+        The name of the environment variable containing the JSON
+        specification.""")
+
+    target = param.String(default=None, doc="""
+        Optional key in the JSON specification dictionary containing the
+        desired parameter values.""")
+
+    def __call__(self, parameterized):
+
+        warnobj = param.main if isinstance(parameterized, type) else parameterized
+        param_class = (parameterized if isinstance(parameterized, type)
+                       else parameterized.__class__)
+
+        p = param.ParamOverrides(self, {})
+        target = p.target if p.target is not None else param_class.__name__
+
+        env_var = os.environ.get(p.varname, None)
+        if env_var is None: return
+
+        spec = json.loads(env_var)
+        if not isinstance(spec, dict):
+            warnobj.warning('JSON parameter specification must be a dictionary.')
+            return
+
+        if target in spec:
+            params = spec[target]
+        else:
+            params = spec
+
+        for name, value in params.items():
+           try:
+               parameterized.set_param(**{name:value})
+           except ValueError as e:
+               if isinstance(parameterized, type):
+                   warnobj.warning(str(e))
+               else:
+                   warnobj.warning(str(e))
```

Comment:
Right..I defined ``warnobj`` conditionally but forgot to make use of that properly.